zebra-chain/src/orchard/note/ciphertexts.rs

```diff
@@ -0,0 +1,133 @@
+use std::{fmt, io};
+
+use crate::serialization::{serde_helpers, SerializationError, ZcashDeserialize, ZcashSerialize};
+
+/// A ciphertext component for encrypted output notes.
+///
+/// Corresponds to the Orchard 'encCiphertext's
+#[derive(Deserialize, Serialize)]
+pub struct EncryptedNote(#[serde(with = "serde_helpers::BigArray")] pub [u8; 580]);
+
+impl fmt::Debug for EncryptedNote {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_tuple("EncryptedNote")
+            .field(&hex::encode(&self.0[..]))
+            .finish()
+    }
+}
+
+// These impls all only exist because of array length restrictions.
```

Comment:
```suggestion
// These impls all only exist because of array length restrictions.
// TODO: use const generics https://github.com/ZcashFoundation/zebra/issues/2042
```